x-pack/plugins/secops/public/components/page/network/network_top_n_flow_table/columns.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiIcon, EuiToolTip } from '@elastic/eui';
+import numeral from '@elastic/numeral';
+import { get, has } from 'lodash/fp';
+import React from 'react';
+import styled from 'styled-components';
+import uuid from 'uuid';
+
+import {
+  NetworkDirectionEcs,
+  NetworkTopNFlowDirection,
+  NetworkTopNFlowEdges,
+  NetworkTopNFlowType,
+} from '../../../../graphql/types';
+import { escapeQueryValue } from '../../../../lib/keury';
+import { networkModel } from '../../../../store';
+import { DragEffects, DraggableWrapper } from '../../../drag_and_drop/draggable_wrapper';
+import { escapeDataProviderId } from '../../../drag_and_drop/helpers';
+import { defaultToEmptyTag, getEmptyTagValue } from '../../../empty_value';
+import { Columns } from '../../../load_more_table';
+import { Provider } from '../../../timeline/data_providers/provider';
+import { AddToKql } from './add_to_kql';
+import * as i18n from './translations';
+
+export const getNetworkTopNFlowColumns = (
+  startDate: number,
+  topNFlowDirection: NetworkTopNFlowDirection,
+  topNFlowType: NetworkTopNFlowType,
+  type: networkModel.NetworkType
+): Array<Columns<NetworkTopNFlowEdges>> => [
+  {
+    name: getIpTitle(topNFlowType),
+    truncateText: false,
+    hideForMobile: false,
+    render: ({ node }) => {
+      const ipAttr = `${topNFlowType}.ip`;
+      const ip: string | null = get(ipAttr, node);
+      const id = escapeDataProviderId(
+        `networkTopNFlow-table-${uuid.v4()}-${topNFlowType}-${topNFlowDirection}-ip-${ip}`
```

Comment:
Clicking on the `Rows` drop down without selecting a new value will result in the drag and drop providers being un-registered and re-registered. You'll need a stable id instead of `uuid.v4()` to prevent this.

![re-register](https://user-images.githubusercontent.com/2946766/53275994-83872580-36ba-11e9-9b14-303e6db73f96.gif)

Reply:
Add a discussion with @FrankHassanabad and we do not need it